src/org/jitsi/impl/neomedia/rtcp/RTCPTCCPacket.java

```diff
@@ -0,0 +1,322 @@
+     * @param packets the set of RTP sequence numbers and their reception
+     * timestamps which this packet is to describe.
+     * @param fbPacketCount the index of this feedback packet, to be used in the
+     * "fb pkt count" field.
+     *
+     * Note that this implementation is not optimized and might not always use
+     * the minimal possible number of bytes to describe a given set of packets.
+     * Specifically, it does take into account that sequence numbers wrap
+     * at 2^16 and fails to pack numbers close to 2^16 with those close to 0.
+     */
+    public RTCPTCCPacket(long senderSSRC, long sourceSSRC,
+                         TreeMap<Integer, Long> packets,
+                         byte fbPacketCount)
+    {
+        super(FMT, RTPFB, senderSSRC, sourceSSRC);
+
+        TreeSet<Map.Entry<Integer, Long>> sequenceNumbers
+            = (TreeSet) packets.entrySet();
+
+        Map.Entry<Integer, Long> first = sequenceNumbers.first();
+        int firstSeq = first.getKey();
+        Map.Entry<Integer, Long> last = sequenceNumbers.last();
+        int packetCount
+            = 1 + RTPUtils.sequenceNumberDiff(last.getKey(), firstSeq);
+
+        byte[] buf = new byte[(packetCount / 7 + 1) * 2 + 8];
+        byte[] deltas = new byte[packetCount * 2];
+        int deltaOff = 0;
+        int off = 0;
+
+        long referenceTime = first.getValue();
+        referenceTime -= referenceTime % 64;
+
+        // Set the 'base sequence number' field
+        off += RTPUtils.writeShort(buf, off, (short) (int) first.getKey());
+
+        // Set the 'packet status count' field
+        off += RTPUtils.writeShort(buf, off, (short) packetCount);
+
+        // Set the 'reference time' field
+        off +=
+            RTPUtils.writeUint24(buf, off,
+                                 (int) ((referenceTime >> 6) & 0xffffff));
+
+        // Set the 'fb pkt count' field. TODO increment
+        buf[off++] = fbPacketCount;
+
+        // Add the packet status chunks. In this first impl we'll just use
+        // status vector chunks (T=1) with two-bit symbols (S=1) as this is
+        // most straightforward to implement.
+        // TODO: optimize for size
+        long nextReferenceTime = referenceTime;
+        off--; // we'll take care of this inside the loop.
+        for (int seqDelta = 0; seqDelta < packetCount; seqDelta++)
+        {
+            // A status vector chunk with two-bit symbols contains 7 packet
+            // symbols
+            if (seqDelta % 7 == 0)
+            {
+                off++;
+                buf[off] = (byte) 0xc0; //T=1, S=1
+            }
+            else if (seqDelta % 7 == 3)
+            {
+                off++;
+                buf[off] = 0;
+            }
+
+            int symbol;
+
+            int seq = (firstSeq + seqDelta) & 0xffff;
+            Long ts = packets.get(seq);
+            if (ts == null)
+            {
+                symbol = 0; // packet not received
```

Comment:
constants for these symbols (packet not received, packet received, small delta, etc.) would make this more self-documenting i think